docs/user-guide/a1-01-matrix-layout.md

```diff
@@ -33,37 +57,40 @@ slang::ISession* session;
+* Host vector interpretation (row or column) - and therefore effective tranform order (column) `m * v` or (row) `v * m`
+* Host matrix memory layout
+* Shader vector interpretation (as determined via `mul(v, m)` or `mul(m, v)`
+* Shader matrix memory layout 
 
-- Make sure the host code fills in matrices in the odering so that translation terms are specified in `m[3], m[7], m[11]` elements.
-- Leave `defaultMatrixLayoutMode` as default value when creating a Slang session, or specify `SLANG_MATRIX_LAYOUT_ROW_MAJOR`.
-- Write `mul(Matrix, Vector)` in shader code to transform `Vector` by `Matrix`.
+The simplest scenario is that they are all `row-major`/`row vector`. In that case code will be most easily portable. NOTE! Currently this requires setting Slangs default matrix layout mode to `row-major`, as it's default is `column-major`. 
 
-**Option 2: using row-major matrix layout, and `mul(v, m)` math convention**
+Also simple but *not* as portable is if they are all `column-major`/`column vector`. It is not portable because some targets (CUDA/C++/CPU) ignore `column-major` matrix layout.
 
-- Make sure the host code fills in matrices so that translations terms are specified in `m[12], m[13], m[14]` elements. Matrices filled in this way are compatible with typical OpenGL applications.
-- Leave `defaultMatrixLayoutMode` as default value when creating a Slang session, or specify `SLANG_MATRIX_LAYOUT_ROW_MAJOR`.
-- Write `mul(Vector, Matrix)` in shader code.
+If there is an inconsistency, then it may be necessary to transpose at the host/kernel boundary.
 
-**Option 3: using column-major matrix layout, and `mul(m, v)` math convention**
+Any flip between `vector interpretation`/`matrix layout` is effectively a transpose, as long as there are an *even* number of `row-major layout`/`row vector interpretation`s then the expected results will be produced. If there is an odd number a transpose will be necessary at the host/kernel boundary.
 
-- Make sure the host code fills in matrices in the odering so that translation terms are specified in `m[12], m[13], m[14]` elements. Matrices filled in this way are compatible with typical OpenGL applications.
-- Set `defaultMatrixLayoutMode` to `SLANG_MATRIX_LAYOUT_COLUMN_MAJOR` when creating a Slang session.
-- Write `mul(Matrix, Vector)` in shader code to transform `Vector` by `Matrix`.
+By this logic...
 
-**Option 4: using column-major matrix layout, and `mul(v, m)` math convention**
+* row vector host interpretation
+* row-major matrix host layout
+* column vector shader interpretation
+* column-major matrix shader layout
 
-- Make sure the host code fills in matrices so that translations terms are specified in `m[3], m[7], m[11]` elements.
-- Set `defaultMatrixLayoutMode` to `SLANG_MATRIX_LAYOUT_COLUMN_MAJOR` when creating a Slang session.
-- Write `mul(Vector, Matrix)` in shader code.
+Will be ok as are 2 (ie *even*) 'row' (one vector interpretation and the other matrix layout).
```

Comment:
I would still prefer we use the original wording that just lists out all viable options a developer can have, so readers can directly refer to them and use it as a checklist for the right set-up.